src/test/java/org/apache/maven/plugin/resources/remote/RemoteResourcesMojoTest.java

```diff
@@ -495,46 +490,66 @@ protected ProcessRemoteResourcesMojo lookupProcessMojo()
-        throws Exception
+                                                                        ArrayList<String> bundles )
+            throws Exception
     {
         final ProcessRemoteResourcesMojo mojo = lookupProcessMojo();
 
-        ArtifactRepository localRepository = mojo.repositorySystem.createLocalRepository( new File( LOCAL_REPO ) );
         DefaultRepositorySystemSession reposession = MavenRepositorySystemUtils.newSession();
         reposession.setLocalRepositoryManager( new SimpleLocalRepositoryManagerFactory()
                 .newInstance( reposession, new LocalRepository( new File( LOCAL_REPO ) ) ) );
 
         DefaultMavenExecutionRequest request = new DefaultMavenExecutionRequest();
-        request.setLocalRepository( localRepository );
-        request = new DefaultMavenExecutionRequest();
         request.setUserProperties( null );
-        request.setLocalRepository( localRepository );
-        request.setGoals(Collections.singletonList( "install" ) );
+        //request.setLocalRepository( reposession.getLocalRepository() );
+        request.setGoals( Collections.singletonList( "install" ) );
         request.setBaseDirectory( project.getBasedir() );
         request.setStartTime( Calendar.getInstance().getTime() );
-        MavenSession session = new MavenSession( getContainer(), reposession, request, new DefaultMavenExecutionResult() );
+
+        // TODO: get rid of this uglyness/legacy
+        MavenArtifactRepository localRepository = new MavenArtifactRepository();
+        localRepository.setId( "local" );
+        localRepository.setUrl( new File( LOCAL_REPO ).toURI().toASCIIString() );
+        localRepository.setLayout( new DefaultRepositoryLayout() );
+        request.setLocalRepository( localRepository );
+        MavenSession session =
+                new MavenSession( getContainer(), reposession, request, new DefaultMavenExecutionResult() );
         session.setProjects( Collections.singletonList( project ) );
 
-        setVariableValueToObject( mojo, "project", project );
         setVariableValueToObject( mojo, "outputDirectory", new File( project.getBuild().getOutputDirectory() ) );
         setVariableValueToObject( mojo, "resourceBundles", bundles );
         setVariableValueToObject( mojo, "mavenSession", session );
-        setVariableValueToObject( mojo, "remoteArtifactRepositories", project.getRemoteArtifactRepositories() );
-        setVariableValueToObject( mojo, "resources", project.getResources() );
         return mojo;
     }
 
     protected ProcessRemoteResourcesMojo lookupProcessMojoWithDefaultSettings( final MavenProject project )
-        throws Exception
+            throws Exception
     {
         return lookupProcessMojoWithSettings( project, new ArrayList<>() );
     }
+
+    private String pathOf( Artifact artifact )
+    {
+        String path = LOCAL_REPO + artifact.getGroupId().replaceAll( "\\.", "/" ) + "/"
```

Comment:
Wouldn't just `replace()` suffice here? We don't need regex.

Reply:
fixed